codegen/masm/intrinsics/i32.masm

```diff
@@ -21,19 +21,18 @@ export.overflowing_add # [b, a]
     # is `a` signed?
     dup.2 exec.is_signed # [is_a_signed, is_b_signed, b, a]
 
+    # do both operands have the same sign?
+    dup.1 eq # [is_same_sign, is_signed, b, a]
```

Comment:
```suggestion
    dup.1 eq # [is_same_sign, is_b_signed, b, a]
```

Reply:
I wanted to make it clear that the purpose of that value after computing `is_same_sign` is to indicate whether `a` and `b` are signed (when `is_same_sign` is true). When `is_same_sign` is false, it's irrelevant whether `is_signed` refers to `a` or `b` specifically, since when the signs differ, overflow is impossible.

That said, if you found it confusing, I'm happy to commit this suggestion. What do you think?

Reply:
Well, after the explanation, it makes sense. Although it still requires more cognitive overhead, I'm ok with it.

Reply:
I'll rewrite the comments here to make it clearer upon first reading - I'd rather not risk it be confusing.